### ISSUE TITLE
History / Diff against depot does not show up on Gluon (Cloud edition) workflow

### ISSUE THREAD

**Issue by L-Naej** [SRombauts/UE4PlasticPlugin#4]
Hello,
Thanks for your work on this plugin!

We've got a problem here with the artist workflow, I mean when only Gluon has been configured on a computer. We are using Gluon with Plastic Cloud Edition. Our Unreal Engine version is 4.14 so we use your last version for 4.14.
When we try to show the history or "diff against depot" a file, nothing happen (no window is opened) and we see this error message in the "Output Log" (in bold is where I replaced our private data):

```
LogSourceControl: **[...]**Content/BP/BP_ProgressionSpline.uasset CanCheckout=1
LogSourceControl: **[...]**Content/BP/BP_ProgressionSpline.uasset CanCheckIn=0
LogSourceControl: ExecuteSynchronousCommand: UpdateStatus
LogSourceControl: IssueCommand: UpdateStatus
LogSourceControl: status (of 1 files, ShouldCheckAllFiles=1, ShouldUpdateHistory=1, ShouldGetOpenedOnly=0, ShouldUpdateModifiedState=0)
LogSourceControl: RunUpdateStatus: 1 files ('**[...]**Content/BP/BP_ProgressionSpline.uasset'...)
LogSourceControl: RunStatus(**[...]**Content/BP/BP_ProgressionSpline.uasset): 2) general case for one or more files in a directory (**[...]**/Content/BP)
LogSourceControl: 'status --wkconfig --noheaders --all --ignored "**[...]**Content/BP"' (in 0.169737s) output:
cs:305@rep:**[our_repository]**@repserver:**[our organisation]**@cloud
Branch /main/Dev@**[our_repository]**@repserver:**[our organisation]**@cloud
LogSourceControl: **[...]**Content/BP/BP_ProgressionSpline.uasset = 2:Controlled
LogSourceControl: 'fileinfo --format="{RevisionChangeset};{RevisionHeadChangeset};{LockedBy};{LockedWhere}" "**[..]**Content/BP/BP_ProgressionSpline.uasset"' (in 0.374391s) output:
125;125;;
LogSourceControl: **[..]**Content/BP/BP_ProgressionSpline.uasset: 125;125 by '' ()
LogSourceControl: 'history --format="{1};{6}" "**[..]**Content/BP/BP_ProgressionSpline.uasset"' (in 0.038052s) output:
125;5822
LogSourceControl:warning: 'log cs:125@rep:@repserver: --xml --encoding="utf-8"' (in 0.026212s) output:
The specified repository couldn't be found: @repserver:.
LogSourceControl:Error: Command 'UpdateStatus' Failed!
```

Any idea of why this happen?
Thanks,
Jean-Noël

**Comment by L-Naej** [comment]
Hi Sébastien,

Thanks for your quick answer. I did not made the link between partial checkin and Gluon. 
Basically what changes with Gluon is that artists can check-in "per file" and choose to not download the entire repository (they can select which directories to download). It also allow them to checkout and lock files, something not possible with Plastic SCM.

They also don't have to bother with branching as Gluon does not allow to branch on its GUI (but you can modify a text file to change the branch it is pointing to). 

Hope it helps!

Regards,
Jean-Noël

**Comment by SRombauts** [comment]
Thanks for taking the time to explain, but I should have been more clear about it; I know quite well what Gluon is for.

My problem is very low level, that is, how to work on this partial checkin mode with the command line interface, what would require modification inside the plugin, what kind of UI would need to be added...

I will need to dive into this when I can dedicate more time to use it a bit, then design the change required and see what it breaks. It will certainly not happen in the coming weeks, though.

**Comment by L-Naej** [comment]
Thanks for the clarification and the kind of ETA! I will keep an eye on this as it would clearly improve our artists workflow.

**Comment by SRombauts** [comment]
Fixed by multiple-commits leading to 8e5453a2873bc69d5b1b94e952384e0a48f02007 Gluon/Partial mode using "cm partial" commands for "checkin" and "update"
